dev/managedservicesplatform/internal/resource/monitoringalertpolicy/monitoringalertpolicy.go

```diff
@@ -0,0 +1,299 @@
+		// No default for this
+	} else {
+		return nil, errors.Newf("invalid service kind %q", config.ServiceKind)
+	}
+
+	if config.ThresholdAggregation.Comparison == "" {
+		config.ThresholdAggregation.Comparison = ComparisonGT
+	}
+
+	if config.ThresholdAggregation.Duration == "" {
+		config.ThresholdAggregation.Duration = "0s"
+	}
+
+	// Optional on Jobs
+	var group_by *[]*string
+	if config.ThresholdAggregation.GroupByField != "" {
+		group_by = pointers.Ptr([]*string{pointers.Ptr(config.ThresholdAggregation.GroupByField)})
+	}
+
+	_ = monitoringalertpolicy.NewMonitoringAlertPolicy(scope,
+		id.TerraformID(config.ID), &monitoringalertpolicy.MonitoringAlertPolicyConfig{
+			Project:     pointers.Ptr(config.ProjectID),
+			DisplayName: pointers.Ptr(config.Name),
+			Documentation: &monitoringalertpolicy.MonitoringAlertPolicyDocumentation{
+				Content:  config.Description,
+				MimeType: pointers.Ptr("text/markdown"),
+			},
+			Combiner: pointers.Ptr("OR"),
+			Conditions: []monitoringalertpolicy.MonitoringAlertPolicyConditions{
+				{
+					DisplayName: pointers.Ptr(config.Name),
+					ConditionThreshold: &monitoringalertpolicy.MonitoringAlertPolicyConditionsConditionThreshold{
+						Aggregations: []monitoringalertpolicy.MonitoringAlertPolicyConditionsConditionThresholdAggregations{
+							{
+								AlignmentPeriod:    pointers.Ptr(config.ThresholdAggregation.Period),
+								PerSeriesAligner:   pointers.Ptr(string(config.ThresholdAggregation.Aligner)),
+								CrossSeriesReducer: pointers.Ptr(string(config.ThresholdAggregation.Reducer)),
+								GroupByFields:      group_by,
+							},
+						},
+						Comparison:     pointers.Ptr(string(config.ThresholdAggregation.Comparison)),
+						Duration:       pointers.Ptr(config.ThresholdAggregation.Duration),
+						Filter:         pointers.Ptr(filter(config)),
+						ThresholdValue: pointers.Float64(config.ThresholdAggregation.Threshold),
+						Trigger: &monitoringalertpolicy.MonitoringAlertPolicyConditionsConditionThresholdTrigger{
+							Count: pointers.Float64(1),
+						},
+					},
+				},
+			},
+			AlertStrategy: &monitoringalertpolicy.MonitoringAlertPolicyAlertStrategy{
+				AutoClose: pointers.Ptr("604800s"),
+			},
+		})
+	return &Output{}, nil
+}
+
+// filter creates the Filter string for a ThresholdAggregation monitoring alert policy
+func filter(config *Config) string {
+	filters := make([]string, 0)
+	for key, val := range config.ThresholdAggregation.Filters {
+		filters = append(filters, fmt.Sprintf(`%s = "%s"`, key, val))
+	}
+
+	// Sort to ensure stable output for testing
+	// This code runs so infreuqently that sort performance is not a concern and there are only a couple elements
+	sort.Strings(filters)
+
+	if config.ServiceKind == spec.ServiceKindService {
+		filters = append(filters, `resource.type = "cloud_run_revision"`)
+		filters = append(filters, fmt.Sprintf(`resource.labels.service_name = "%s"`, config.ServiceName))
```

Comment:
nit - might be a bit easier to read all in one, given we add these together:

```suggestion
		filters = append(filters,
			`resource.type = "cloud_run_revision"`)
			fmt.Sprintf(`resource.labels.service_name = "%s"`, config.ServiceName))
```

same for the next block :)